lib/linguist/strategy/modeline.rb

```diff
@@ -144,6 +145,9 @@ def self.call(blob, _ = nil)
         return [] if blob.symlink?
 
         header = blob.first_lines(SEARCH_SCOPE).join("\n")
+        # Return early for Vimball files as their modeline will not reflect their filetype.
+        return [] if header.include?("UseVimball")
+
```

Comment:
What do you think of this @Alhadis?

Reply:
Looks great! (I was about to suggest that a regression test be added, but remembered that most samples are implicitly used as fixtures anyway... 😅)